tests/scripts/all.sh

```diff
@@ -4339,6 +4339,47 @@ component_build_aes_aesce_armcc () {
     armc6_build_test "-O1 --target=aarch64-arm-none-eabi -march=armv8-a+crypto"
 }
 
+support_build_sha_armce() {
+    # clang >= 4 is required to build with SHA extensions
+    ver="$(clang --version|grep version|sed -E 's#.*version ([0-9]+).*#\1#')"
+    [ "${ver}" -ge 4 ]
+}
+
+component_build_sha_armce () {
+    # Test variations of SHA256 Armv8 crypto extensions
+    scripts/config.py unset MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_IF_PRESENT
+    scripts/config.py set MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_ONLY
+
+    msg "MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_ONLY clang, aarch64"
+    make -B library/sha256.o CC=clang CFLAGS="--target=aarch64-linux-gnu -march=armv8-a"
+
+    msg "MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_ONLY clang, arm"
+    make -B library/sha256.o CC=clang CFLAGS="--target=arm-linux-gnueabihf -mcpu=cortex-a72+crypto -marm"
+
+    # test the deprecated form of the config option
+    scripts/config.py unset MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_ONLY
+    scripts/config.py set MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY
+
+    msg "MBEDTLS_SHA256_USE_ARMV8_A_CRYPTO_ONLY clang, thumb"
```

Comment:
msg is wrong, since we're testing `_A64_` option

Do we want to check that we get the expected instruction(s) in the output? i.e. that the detection code hasn't decided we're not capable of SHA256 CE, and disabled it?

Reply:
Ideally, but that's not straightforward to do. For native arch we can look at the output of `programs/test/selftest`; for non-native we'd have to run objdump or something.

Out of scope for this PR (I have manually checked it).

Reply:
Since we're only building `sha256.o`, we could just add `-S` to the compiler flags `make -B library/sha256.o CC=clang CFLAGS="--target=arm-linux-gnueabihf -mcpu=cortex-a72+crypto -marm -std=c99"` and then `grep -q sha256h2.32.*q.*q.*q library/sha256.o`